docs/structured-streaming-kafka-integration.md

```diff
@@ -622,6 +626,10 @@ a ```null``` valued key column will be automatically added (see Kafka semantics
 how ```null``` valued key values are handled). If a topic column exists then its value
 is used as the topic when writing the given row to Kafka, unless the "topic" configuration
 option is set i.e., the "topic" configuration option overrides the topic column.
+If a partition column is not specified then the partition is calculated by the Kafka producer
+(using ```org.apache.kafka.clients.producer.internals.DefaultPartitioner```).
+This can be overridden in Spark by setting the ```kafka.partitioner.class``` option.
```

Comment:
Yes, exactly. But this is `KafkaProducer` standard behaviour: 
- it uses `ProducerRecord` partition field. if `null`, fall backs to:
- `kafka.partitioner.class` provided. If not set:
- use default partitioner.

I don't believe we need a test for this (otherwise we would be testing Kafka API) but maybe we should explicitly state it in the doc.

Reply:
What I would like to test here is that Spark parameterizes producer properly.
These testing points has nothing to do with producer internals:
* `kafka.partitioner.class` config reaches producer instances and takes effect
* Partition field is set under some circumstances

Reply:
> Any config that starts with kafka. will be passed down to the producer, actually.

There are exceptions, please see them either in the doc or in the code. If such thing is not checked by a test then it can be broken easily. Just a personal thought: if something is not covered in a test later it's super hard to find out it's a bug or a feature...

> Don't the other test I added prove this point already?

If we add the mentioned simplified test then you're right it will cover it.

Reply:
Do we still need test 1 (`kafka.partitioner.class` overrides default partitioner)?